File: questions/1015. Smallest Integer Divisible by K/smallest.java
/*
1015. Smallest Integer Divisible by K
Medium

Given a positive integer K, you need to find the length of the smallest positive integer N such that N is divisible by K, and N only contains the digit 1.

Return the length of N. If there is no such N, return -1.

Note: N may not fit in a 64-bit signed integer.
*/
class Solution {
    public int smallestRepunitDivByK(int k) {
        int d = k%10;
		if(!(d== 1 || d == 3 || d == 7 || d==9)) return -1;
		HashSet<Integer> set = new HashSet();
		int prev = 0;
		for(int i = 1; i < k + 1; i++){
			int remain = (prev *10+ 1)%k;
			if(remain == 0){
				return i;
			}
			if(set.contains(remain)) return -1;
			prev = remain;
			set.add(remain);
		}
		return -1;        
    }
}

File: questions/1015. Smallest Integer Divisible by K/smallest.py
class Solution:
    def smallestRepunitDivByK(self, k: int) -> int:
        if k%10 not in {1,3,7,9}:
            return -1
        s, prev = set(), 0
        for i in range(1,k + 1):
            prev = (prev*10+1)%k
            if not prev:
                return i
            if prev in s:
                return -1
            s.add(prev)
        return -1
